ambry-replication/src/main/java/com.github.ambry.replication/BlobIdTransformer.java

```diff
@@ -132,11 +138,9 @@ public void warmup(List<MessageInfo> messageInfos) throws Exception {
    * @param newKey the new StoreKey
    * @param oldMessageInfo the {@link MessageInfo} of the message being transformed
    * @return new Message message
-   * @throws IOException
-   * @throws MessageFormatException
+   * @throws Exception
    */
-  private Message newMessage(InputStream inputStream, StoreKey newKey, MessageInfo oldMessageInfo)
-      throws IOException, MessageFormatException {
+  private Message newMessage(InputStream inputStream, StoreKey newKey, MessageInfo oldMessageInfo) throws Exception {
```

Comment:
remind me why this method is called if `newKey.equals(oldMessageInfo.getStoreKey())` instead of using `message` as is?

Reply:
BlobId compareTo for version 3 and higher only compares the UUID which can be problematic for our purposes